src/node_webstorage.cc

```diff
@@ -0,0 +1,661 @@
+
+  if (storage->Store(property, value)) {
+    info.GetReturnValue().Set(value);
+  }
+}
+
+static void StorageQuery(Local<Name> property,
+                         const PropertyCallbackInfo<Integer>& info) {
+  if (!ShouldIntercept(property, info)) {
+    return;
+  }
+
+  Storage* storage;
+  ASSIGN_OR_RETURN_UNWRAP(&storage, info.Holder());
+  Local<Value> result = storage->Load(property);
+  if (!result.IsEmpty()) {
+    info.GetReturnValue().Set(0);
+  }
+}
+
+static void StorageDeleter(Local<Name> property,
+                           const PropertyCallbackInfo<Boolean>& info) {
+  Storage* storage;
+  ASSIGN_OR_RETURN_UNWRAP(&storage, info.Holder());
+
+  if (storage->Remove(property)) {
+    info.GetReturnValue().Set(true);
+  }
+}
+
+static void StorageEnumerator(const PropertyCallbackInfo<Array>& info) {
+  Storage* storage;
+  ASSIGN_OR_RETURN_UNWRAP(&storage, info.Holder());
+  info.GetReturnValue().Set(storage->Enumerate());
+}
+
+static void StorageDefiner(Local<Name> property,
+                           const PropertyDescriptor& desc,
+                           const PropertyCallbackInfo<Value>& info) {
+  Storage* storage;
+  ASSIGN_OR_RETURN_UNWRAP(&storage, info.Holder());
+
+  if (desc.has_value()) {
+    return StorageSetter(property, desc.value(), info);
+  }
+}
+
+static void IndexedStorageGetter(uint32_t index,
+                                 const PropertyCallbackInfo<Value>& info) {
+  Environment* env = Environment::GetCurrent(info);
+  StorageGetter(Uint32ToName(env->context(), index), info);
+}
+
+static void IndexedStorageSetter(uint32_t index,
+                                 Local<Value> value,
+                                 const PropertyCallbackInfo<Value>& info) {
+  Environment* env = Environment::GetCurrent(info);
+  StorageSetter(Uint32ToName(env->context(), index), value, info);
+}
+
+static void IndexedStorageQuery(uint32_t index,
+                                const PropertyCallbackInfo<Integer>& info) {
+  Environment* env = Environment::GetCurrent(info);
+  StorageQuery(Uint32ToName(env->context(), index), info);
+}
+
+static void IndexedStorageDefiner(uint32_t index,
+                                  const PropertyDescriptor& desc,
+                                  const PropertyCallbackInfo<Value>& info) {
+  Environment* env = Environment::GetCurrent(info);
+  StorageDefiner(Uint32ToName(env->context(), index), desc, info);
+}
+
+static void IndexedStorageDeleter(uint32_t index,
+                                  const PropertyCallbackInfo<Boolean>& info) {
+  Environment* env = Environment::GetCurrent(info);
+  StorageDeleter(Uint32ToName(env->context(), index), info);
+}
+
+static void StorageLengthGetter(const FunctionCallbackInfo<Value>& info) {
+  Storage* storage;
+  ASSIGN_OR_RETURN_UNWRAP(&storage, info.This());
+  info.GetReturnValue().Set(storage->Length());
+}
+
+static void StorageLengthSetter(const FunctionCallbackInfo<Value>& info) {
+  info.GetReturnValue().SetUndefined();
```

Comment:
Is this meant to be a non-op? If so, a comment here explaining that would be helpful.